internal/worker/clienterrors/errors.go

```diff
@@ -58,6 +60,70 @@ func (e *Error) String() string {
 	return fmt.Sprintf("Code: %d, Reason: %s, Details: %v", e.ID, e.Reason, e.Details)
 }
 
+func ensureNoErrs(v reflect.Value) error {
+	switch v.Kind() {
+	case reflect.Interface:
+		if errIf, ok := v.Interface().(error); ok {
+			if errIf != nil {
+				return fmt.Errorf("%v", errIf.Error())
+			}
+		}
+	case reflect.Ptr:
+		if err := ensureNoErrs(v.Elem()); err != nil {
+			return err
+		}
+	case reflect.Struct:
+		for i := 0; i < v.NumField(); i++ {
+			if err := ensureNoErrs(v.Field(i)); err != nil {
+				return err
```

Comment:
@mvo5 Do I get this right, that only the first will be returned (same for array and map)? Is this on purpose?

Reply:
Yes, only the first is detected right now. We can change that of course, I'm not sure if it's worth it (also I'm not sure this entire PR is the right approach, it's nice in the sense that it does detect the issue now but it still will not catch other cases where we pass something to `Details interface{}` that is in fact an interface that is not properly serializable to json. 

I still wonder if we a better appraoch would be:
a) merge https://github.com/osbuild/osbuild-composer/pull/4145 to fix the immedidate issue
b) change `Details inteface{}` -> `Details string` or `Details []string` or `Details json.RawMessage` (the later would force the caller to serialize).

Reply:
Not sure where the idea comes from that Details should be flexible but then it would be a) and b) with `json.RawMessage`
Starting with #4145 for now sound perfect 😏

Reply:
it's probably worth to salvage (some of) the test cases from this PR but otherwise I'm fine closing it (or closing it after (a) and (b) got done).